From 51eb232e898ce08de574a064ddde53872c41aec0 Mon Sep 17 00:00:00 2001
From: Josh Poimboeuf <jpoimboe@redhat.com>
Date: Thu, 4 Sep 2014 18:54:42 -0400
Subject: [kernel] trace: Remove function_trace_stop and HAVE_FUNCTION_TRACE_MCOUNT_TEST

Message-id: <376eaf330b864fc53c218ddb6517ee894e508ece.1409856554.git.jpoimboe@redhat.com>
Patchwork-id: 91259
O-Subject: [RHEL7.1 PATCH 23/23] tracing: Remove function_trace_stop and HAVE_FUNCTION_TRACE_MCOUNT_TEST
Bugzilla: 1113829
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1113829

commit 3a636388bae8390d23f31e061c0c6fdc14525786
Author: Steven Rostedt (Red Hat) <rostedt@goodmis.org>
Date:   Thu Jun 26 11:24:52 2014 -0400

    tracing: Remove function_trace_stop and HAVE_FUNCTION_TRACE_MCOUNT_TEST

    All users of function_trace_stop and HAVE_FUNCTION_TRACE_MCOUNT_TEST have
    been removed. We can safely remove them from the kernel.

    Reviewed-by: Masami Hiramatsu <masami.hiramatsu.pt@hitachi.com>
    Signed-off-by: Steven Rostedt <rostedt@goodmis.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/ftrace.h b/include/linux/ftrace.h
index ee4b90c..d0ab246 100644
--- a/include/linux/ftrace.h
+++ b/include/linux/ftrace.h
@@ -116,8 +116,6 @@ struct ftrace_ops {
 #endif
 };
 
-extern int function_trace_stop;
-
 /*
  * Type of the current tracing.
  */
diff --git a/kernel/trace/Kconfig b/kernel/trace/Kconfig
index 015f85a..50a6097 100644
--- a/kernel/trace/Kconfig
+++ b/kernel/trace/Kconfig
@@ -29,11 +29,6 @@ config HAVE_FUNCTION_GRAPH_FP_TEST
  help
    See Documentation/trace/ftrace-design.txt
 
-config HAVE_FUNCTION_TRACE_MCOUNT_TEST
- bool
- help
-   See Documentation/trace/ftrace-design.txt
-
 config HAVE_DYNAMIC_FTRACE
  bool
  help
diff --git a/kernel/trace/ftrace.c b/kernel/trace/ftrace.c
index ba601f0..465d165 100644
--- a/kernel/trace/ftrace.c
+++ b/kernel/trace/ftrace.c
@@ -80,9 +80,6 @@ static struct ftrace_ops ftrace_list_end __read_mostly = {
 int ftrace_enabled __read_mostly;
 static int last_ftrace_enabled;
 
-/* Quick disabling of function tracer. */
-int function_trace_stop __read_mostly;
-
 /* Current function tracing op */
 struct ftrace_ops *function_trace_op __read_mostly = &ftrace_list_end;
 
-- 
1.7.1